.github/workflows/build-and-rebuild.yaml

```diff
@@ -105,15 +114,23 @@ jobs:
         env:
           LOG_LEVEL: WARNING # Set log level to WARNING
 
+  handle-workflow-dispatch:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'workflow_dispatch'
+    outputs:
+      recipe: ${{ github.event.inputs.recipes }}
+    steps:
+      - name: Set recipes output
+        run: echo "recipes=${{ github.event.inputs.recipes }}" >> "$GITHUB_OUTPUT"
+
   build-and-rebuild-recipes:
-    needs: generate-recipes
+    needs: [generate-recipes, handle-workflow-dispatch]
     name: ${{ matrix.recipe_name }}-${{ matrix.os }}
-    if: fromJson(needs.generate-recipes.outputs.recipe)[0] != null
+    if: needs.generate-recipes.result == 'success' || needs.handle-workflow-dispatch.result == 'success'
```

Comment:
I think you still need this thing `fromJson(needs.generate-recipes.outputs.recipe)[0] != null` otherwise it will fail because input is empty

Reply:
Updated it